Feature/core 1063 update did spec (#3)
evannetwork/evan.network-DID-method-specification: evan_did_method_spec.md

An evan.network DID is depending on the type of the identity. There are three types of identities:<br>

**Self-sovereign Identity (SSI)**<br>

> Self-sovereign identities (SSIs) can either be a company, a device, or a user. These identities are created, owned, and managed by a person via an externally owned account/private key and have their own means of authentication and signing and can issue verified credentials. Changes to the DID document itself can only be performed by the owner or another SSI referenced via the `controller` property of the DID document. Upon creation the `controller` property is set to the owner of the identity so that the owner has the full control over the identity. Alongside this, each DID document holds a proof which is generated using the owner's key. SSIs fully feature key revocation and key rotation.

Reviewer: ```suggestion
Self-sovereign identities (SSIs) can either be a company, a device, or a user. These identities are created, owned, and managed by a person via an externally owned account/private key and have their own means of authentication and signing and can issue verifiable credentials (VCs). Changes to the DID document itself can only be performed by the owner or another SSI referenced via the `controller` property of the DID document. Each DID document holds a proof which is generated using the owner's key. SSIs fully feature key revocation and key rotation.
```

Reviewer: Dropped the "Upon creation the `controller` property is set to the owner..." because that is not the case for us

Reviewer 2: >  These identities are created, owned, and managed by a person via an externally owned account/private key and have their own means of authentication and signing and can issue verifiable credentials (VCs)

This sentence has a lot of 'and's which make it hard to understand. Maybe we can split that in 2 or 3 short sentences or steps. See also: http://www.cypressmedia.net/articles/article/26/six_principles_of_technical_writing#sentences

The following sample shows an asset identity's DID document. The asset has no `authentication` means and only points to its owner's SSI which is delegated to authenticate in this asset identity's name, as well as apply changes to the DID document (via the `controller` property).

```js
{ '@context': 'https://w3id.org/did/v1',
  id:

>    'did:evan:testcore:0x8dfe627ab531a9b9b23910ca691df53dfb13c1b157777360d53d68172aa30ba9',

Reviewer: I cannot resolve this DID at the [resolver](https://testcore.evan.network/did/did:evan:testcore:0x8dfe627ab531a9b9b23910ca691df53dfb13c1b157777360d53d68172aa30ba9). Does ist exist?

You must be either the owner or the controller of the identity to apply modifications and updates can only be performed by SSIs.

```javascript
{ '@context': 'https://w3id.org/did/v1',
  id:
   'did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906',
  publicKey:
   [ { id:
        'did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#key-1',
       type: 'Secp256k1VerificationKey2018',
       controller:
        'did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906',
       ethereumAddress: '0xcd5e1dbb5552c2baa1943e6b5f66d22107e9c05c' } ],
  authentication:
   [ 'did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#key-1' ],
  created: '2020-03-19T09:56:33.808Z',
  updated: '2020-03-19T09:56:34.563Z',
  service:
   [ { id:
        'did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#randomService',
       type: 'randomService-575000253',
       serviceEndpoint: 'https://openid.example.com/575000253' } ] 

> }

Reviewer: Could use a proof property

Just use the document from [here](https://testcore.evan.network/did/did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906)

You must be either the owner or the controller of the identity to apply modifications and updates can only be performed by SSIs.

```javascript
{
  "did": {
    "@context": "https://w3id.org/did/v1",
    "id": "did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906",
    "publicKey": [
      {
        "id": "did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#key-1",
        "type": "Secp256k1VerificationKey2018",
        "controller": "did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906",
        "ethereumAddress": "0xcd5e1dbb5552c2baa1943e6b5f66d22107e9c05c"
      }
    ],
    "authentication": [
      "did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#key-1"
    ],
    "created": "2020-03-24T08:24:54.398Z",
    "updated": "2020-03-24T08:24:57.546Z",
    "proof": {
      "type": "EcdsaPublicKeySecp256k1",
      "created": "2020-03-24T08:24:57.556Z",
      "proofPurpose": "assertionMethod",
      "verificationMethod": "did:evan:testcore:0x0d87204c3957d73b68ae28d0af961d3c72403906#key-1",

>       "jws": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NkstUiJ9.eyJpYXQiOjE1ODUwMzgyOTcsImRpZERvY3VtZW50Ijp7IkBjb250ZXh0IjoiaHR0cHM6Ly93M2lkLm9yZy9kaWQvdjEiLCJpZCI6ImRpZDpldmFuOnRlc3Rjb3JlOjB4MGQ4NzIwNGMzOTU3ZDczYjY4YWUyOGQwYWY5NjFkM2M3MjQwMzkwNiIsInB1YmxpY0tleSI6W3siaWQiOiJkaWQ6ZXZhbjp0ZXN0Y29yZToweDBkODcyMDRjMzk1N2Q3M2I2OGFlMjhkMGFmOTYxZDNjNzI0MDM5MDYja2V5LTEiLCJ0eXBlIjoiU2VjcDI1NmsxVmVyaWZpY2F0aW9uS2V5MjAxOCIsImNvbnRyb2xsZXIiOiJkaWQ6ZXZhbjp0ZXN0Y29yZToweDBkODcyMDRjMzk1N2Q3M2I2OGFlMjhkMGFmOTYxZDNjNzI0MDM5MDYiLCJldGhlcmV1bUFkZHJlc3MiOiIweGNkNWUxZGJiNTU1MmMyYmFhMTk0M2U2YjVmNjZkMjIxMDdlOWMwNWMifV0sImF1dGhlbnRpY2F0aW9uIjpbImRpZDpldmFuOnRlc3Rjb3JlOjB4MGQ4NzIwNGMzOTU3ZDczYjY4YWUyOGQwYWY5NjFkM2M3MjQwMzkwNiNrZXktMSJdLCJjcmVhdGVkIjoiMjAyMC0wMy0yNFQwODoyNDo1NC4zOThaIiwidXBkYXRlZCI6IjIwMjAtMDMtMjRUMDg6MjQ6NTcuNTQ2WiIsInNlcnZpY2UiOlt7ImlkIjoiZGlkOmV2YW46dGVzdGNvcmU6MHgwZDg3MjA0YzM5NTdkNzNiNjhhZTI4ZDBhZjk2MWQzYzcyNDAzOTA2I3JhbmRvbVNlcnZpY2UiLCJ0eXBlIjoicmFuZG9tU2VydmljZS00MDA4MTY0NDIiLCJzZXJ2aWNlRW5kcG9pbnQiOiJodHRwczovL29wZW5pZC5leGFtcGxlLmNvbS80MDA4MTY0NDIifV19LCJpc3MiOiJkaWQ6ZXZhbjp0ZXN0Y29yZToweDBkODcyMDRjMzk1N2Q3M2I2OGFlMjhkMGFmOTYxZDNjNzI0MDM5MDYifQ.B4AWY0qTnWmrWa9KJjtDUbWMcGXNITFCJY9xcYhJ5XZi7zY6EJCcVBPwKNBp25mJ3A1z43g5vgaykJNj7XnsQQA"

Reviewer: ```suggestion
      "jws": "..."
```
Suggestion to remove the JWS to not distract from the services here, because it is a lot. But you can keep it if you like it better that way
